langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/TypeParamTest.java

```diff
@@ -67,13 +67,9 @@ public void testTypeParamNegative() {
         BAssertUtil.validateError(result, err++, "incompatible types: expected 'int', found 'float'", 119, 21);
         BAssertUtil.validateError(result, err++, "incompatible types: expected 'byte', found 'int'", 122, 31);
         BAssertUtil.validateError(result, err++, "incompatible types: expected 'byte', found 'int'", 125, 26);
-
-        // Disabled due to https://github.com/ballerina-platform/ballerina-lang/issues/22137
-        // BAssertUtil.validateError(result, err++, "incompatible types: expected '(int|string)', found 'int'", 130,
-        // 14);
-        // BAssertUtil.validateError(result, err++, "incompatible types: expected '(int|string)', found 'float'", 131,
-        //                           24);
-
+        BAssertUtil.validateError(result, err++, "incompatible types: expected '(int|string)', found 'boolean'", 130,
+                                  14);
+        BAssertUtil.validateError(result, err++, "incompatible types: expected '(int|string)', found 'float'", 131, 24);
```

Comment:
Shall we add a few more negative test cases with incompatible unions?
Such as on (int[]|string[]|float), (int[]|map<int>)

Reply:
Don't think code visit this logic for negative cases? It gives an `undefined function`.
I added some different test cases ec685a4fd8a1d34e0a93b9eb724c0c95f3d51a9c
Any suggestions?